Lib/test/test_htmlparser.py

```diff
@@ -787,5 +789,17 @@ def test_weird_chars_in_unquoted_attribute_values(self):
                             ('starttag', 'form',
                                 [('action', 'bogus|&#()value')])])
 
+
+class TestInheritance(unittest.TestCase):
+
+    @patch("_markupbase.ParserBase.__init__")
+    @patch("_markupbase.ParserBase.reset")
+    def test_init(self, super_init_method, super_reset_method):
+        with patch('_markupbase.ParserBase') as parser_base:
+            a = html.parser.HTMLParser()
```

Comment:
Here I would use a subclass of `HTMLParser`, since that's what most people will use in real code.  We can reuse the `EventCollector` defined above.
```suggestion
            EventCollector()
```